artemis-commons/src/test/java/org/apache/activemq/artemis/utils/Wait.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.utils;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Assert;
+
+/**
+ * Utility adapted from: org.apache.activemq.util.Wait
+ */
+public class Wait {
```

Comment:
Can we avoid duplicating this, and simply make it more shareable between modules.

Reply:
@brusdev you did this on top of my previous commit, I had the Wait there as a hack, and I intended to fix this before merging.

We need to fix the Wait here.

(I think you should have kept my commit as a reference of my ownership though and changed it on top.. but no big deal)

Reply:
this is a separate issue. It's about the artemis-junit project duplicating some stuff.

I'm removing the dependency on artemis-junit, and I'm dealing with this on a separate PR. I am merging this and i will send a new PR.